File: TechVault.API/Models/Product.cs
namespace TechVault.API.Models;

public class Product
{
    public int Id { get; set; }
    public int CategoryId { get; set; }
    public string Name { get; set; } = null!;
    public string Slug { get; set; } = null!;
    public string? ShortDescription { get; set; }
    public string? Description { get; set; }
    public string Sku { get; set; } = null!;
    public decimal Price { get; set; }
    public decimal? CompareAtPrice { get; set; }
    public int StockQuantity { get; set; }
    public string? ImageUrl { get; set; }
    public string? Brand { get; set; }
    public bool IsPublished { get; set; }
    public DateTime CreatedAtUtc { get; set; }
    public DateTime? UpdatedAtUtc { get; set; }

    public Category Category { get; set; } = null!;
    public ICollection<ProductTag> ProductTags { get; set; } = new List<ProductTag>();
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public ICollection<CartItem> CartItems { get; set; } = new List<CartItem>();
    public ICollection<Review> Reviews { get; set; } = new List<Review>();
}

File: techvault-client/src/types/product.ts
/** Matches TechVault.API `ProductDto` (camelCase JSON). */
export type ProductCategoryDto = {
  id: number
  name: string
  slug: string
}

/** Matches `ProductListItemDto` from paged product lists. */
export type ProductListItem = {
  id: number
  name: string
  slug: string
  price: number
  compareAtPrice: number | null
  imageUrl: string | null
  brand: string | null
  categoryName: string
  categorySlug: string
  stockQuantity: number
  isPublished: boolean
}

export type PagedProductsResponse = {
  items: ProductListItem[]
  totalCount: number
  page: number
  pageSize: number
  totalPages: number
}

export type ProductDetail = {
  id: number
  name: string
  slug: string
  sku: string
  description: string | null
  price: number
  stock: number
  images: string[]
  category: ProductCategoryDto
  specs: Record<string, string>
}
